spec/unit/client-request-cache.spec.js

```diff
@@ -0,0 +1,120 @@
+const ClientRequestCache = require("../../lib/components/client-request-cache");
+const Promise = require("bluebird");
+
+describe("ClientRequestCache", function() {
+    describe("constructor", function() {
+        it("Can construct", function() {
+            const crc = new ClientRequestCache(50, 1, () => { });
+            expect(crc.ttl).toBe(50);
+            expect(crc.maxSize).toBe(1);
+            expect(crc.requestFunc).toBeDefined();
+            expect(crc._requestContent).toBeDefined();
```

Comment:
Checking internal properties feels wrong

Reply:
We can probably assume they got defined if it didn't throw I guess.